agent/http.go

```diff
@@ -231,13 +232,12 @@ func (s *HTTPServer) wrap(handler func(resp http.ResponseWriter, req *http.Reque
 
 		handleErr := func(err error) {
 			s.agent.logger.Printf("[ERR] http: Request %s %v, error: %v from=%s", req.Method, logURL, err, req.RemoteAddr)
-			code := http.StatusInternalServerError // 500
-			errMsg := err.Error()
-			if strings.Contains(errMsg, "Permission denied") || strings.Contains(errMsg, "ACL not found") {
-				code = http.StatusForbidden // 403
+			switch {
+			case acl.IsErrPermissionDenied(err) || acl.IsErrNotFound(err):
+				http.Error(resp, err.Error(), http.StatusForbidden) // 403
+			default:
+				http.Error(resp, err.Error(), http.StatusInternalServerError) // 500
 			}
-			resp.WriteHeader(code)
```

Comment:
This technically changes the interface a little by adding an extra header and a trailing newline after the message, but `http.Error()` seems like a nicer way to do what we are doing here. It's probably ok as those changes are pretty trivial.